package/Dockerfile

```diff
@@ -1,17 +1,26 @@
-FROM registry.suse.com/bci/bci-base:15.6 AS builder
+FROM registry.suse.com/bci/bci-base:15.6 AS base
 RUN sed -i 's/^CREATE_MAIL_SPOOL=yes/CREATE_MAIL_SPOOL=no/' /etc/default/useradd
 RUN useradd --uid 1007 aks-operator
 
+FROM registry.suse.com/bci/golang:1.23 AS builder
+WORKDIR /app
+COPY go.mod go.sum ./
+RUN go mod download && go mod verify
+COPY ./controller ./controller
+COPY ./pkg ./pkg
+COPY ./main.go ./main.go
+RUN CGO_ENABLED=0 GOOS=linux go build -o /aks-operator
```

Comment:
Due to the cross compilation, it may be sensible to [verify](https://github.com/rancher/cis-operator/blob/main/package/Dockerfile#L60) the generated binary is valid for the target architecture.

Reply:
Good idea :+1:  I've updated the PR to use cross-compilation helpers.